### DISCUSSION TITLE
Updating CodeProject

### DISCUSSION THREAD

**Discussion by fotingo** [codeproject/CodeProject.AI-Server#230] [Q&A]
It's been already twice in a row that I try to update CodeProject to whatever the newest update is and it gets stuck in installing, then giving an error "Failed Install". 

I always select the option to not uninstall the current module and that seems to give an issue. To fix this, I have to uninstall CodeProject and then install the newest version. Why is this?

**Comment by dannyeverest** [comment]
Please include more info. I'm assuming you're on windows. 

I've heard you need to delete the /appdata/ cpai files on windows manually... not sure if that helps but good luck. 

Also you could run it in docker desktop. Will most likely work great.

If you are unfamiliar with docker, prepare to spend a little time. Docker is epic. Seems pretty fast and lean.
